include/xsimd/config/xsimd_config.hpp

```diff
@@ -128,6 +128,9 @@
  * Set to 1 if AVX512F is available at compile-time, to 0 otherwise.
  */
 #ifdef __AVX512F__
+#if defined(__clang__) && __clang_major__ >= 6
+    #define XSIMD_WITH_AVX512F 1
+#else
     // AVX512 instructions are supported starting with gcc 6
     // see https://www.gnu.org/software/gcc/gcc-6/changes.html
     #if defined(__GNUC__) && __GNUC__ < 6
```

Comment:
clang also defines `__GNUC__`, but looks it's fixed to 4, even for latest clang-12 [1]. So avx512 is always disabled when build xsimd with clang.
Add explicit clang checking to fix the issue.
[1] https://godbolt.org/z/n6cEexPfn